src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1375,19 +1372,50 @@ bool SQLStorage::dbMigrate() {
     return false;
   }
 
-  if (schema_num_version > current_schema_version) {
-    LOG_ERROR << "Only forward migrations are supported. You cannot migrate to an older schema.";
-    return false;
-  }
-
   SQLite3Guard db = dbConnection();
-  for (int32_t k = schema_num_version + 1; k <= current_schema_version; k++) {
-    if (db.exec(schema_migrations.at(static_cast<size_t>(k)), nullptr, nullptr) != SQLITE_OK) {
-      LOG_ERROR << "Can't migrate db from version " << (k - 1) << " to version " << k << ": " << db.errmsg();
-      return false;
+  if (schema_version != DbVersion::kEmpty && schema_num_version > current_schema_version) {
```

Comment:
I'd suggest making two sub-methods for backward and forward migration, it's a bit heavy on flow-control right now.